pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -151,6 +152,11 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{Requeue: requeue, RequeueAfter: requeueDelayOnError}, err
 	}
 
+	// Temporarily pause reconcile
+	for r.pause {
+		time.Sleep(10 * time.Millisecond)
+	}
```

Comment:
This approach relies on timing and multiple goroutines. This looks rather fragile to me.

As you have to modify the production code anyway (in this case I really don't see an alternative), why not add some callback mechanism here? Then you can reliably execute some additional test code in the middle of the reconcile.

Reply:
Sounds good. I will change to callback.

Reply:
Done. Reconcile hooks are used to validate this case.